simpleclient_eclipselink/src/main/java/io/prometheus/client/eclipselink/EclipseLinkStatisticsCollector.java

```diff
@@ -0,0 +1,744 @@
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import org.eclipse.persistence.sessions.Session;
+import org.eclipse.persistence.sessions.SessionProfiler;
+import org.eclipse.persistence.tools.profiler.PerformanceMonitor;
+
+/**
+ * Collect metrics from one or more EclipseLink Session instances.
+ * <p>
+ * Usage example for a single session:
+ * <pre>
+ * new EclipseLinkStatisticsCollector(session, "name").register();
+ * </pre>
+ * Usage example for multiple sessions:
+ * <pre>
+ * new EclipseLinkStatisticsCollector()
+ *     .add(session1, "name1")
+ *     .add(session2, "name2")
+ *     .register();
+ * </pre>
+ * Session instance can be obtained from EntityManager instance like this:
+ * <pre>
+ * Session session = ((JpaEntityManager)entityManager.getDelegate()).getSession();
+ * </pre>
+ *
+ * @author Viktoriia Bakalova (based on HibernateStatisticsCollector from Christian Kaltepoth);
+ * </p>
+ */
+public class EclipseLinkStatisticsCollector extends Collector {
+
+  private static final List<String> LABEL_NAMES = Arrays.asList("session_name");
+  private final Map<String, Session> sessions = new ConcurrentHashMap<String, Session>();
+
+  /**
+   * Creates an empty collector. If you use this constructor, you have to add one or more session
+   * sessions to the collector by calling the {@link #add(Session, String)} method.
+   */
+  public EclipseLinkStatisticsCollector() {
+  }
+
+  /**
+   * Creates a new collector for the given session. Calling this constructor has the same effect as
+   * creating an empty collector and adding the session using {@link #add(Session, String)}.
+   *
+   * @param session The EclipseLink Session to collect metrics for
+   * @param name A unique name for this Session
+   */
+  public EclipseLinkStatisticsCollector(Session session, String name) {
+    add(session, name);
+  }
+
+  /**
+   * Registers an EclipseLink Session with this collector.
+   *
+   * @param session The EclipseLink Session to collect metrics for
+   * @param name A unique name for this Session
+   * @return Returns the collector
+   */
+  public EclipseLinkStatisticsCollector add(Session session, String name) {
+    sessions.put(name, session);
+    return this;
+  }
+
+  @Override
+  public List<MetricFamilySamples> collect() {
+    List<MetricFamilySamples> metrics = new ArrayList<MetricFamilySamples>();
+    metrics.addAll(getCounters());
+    metrics.addAll(getTimers());
+    return metrics;
+  }
+
+  @Override
+  public <T extends Collector> T register(CollectorRegistry registry) {
+    if (sessions.isEmpty()) {
+      throw new IllegalStateException("You must register at least one Session.");
+    }
+    return super.register(registry);
+  }
+
+  private List<MetricFamilySamples> getCounters() {
+    return Arrays.<MetricFamilySamples>asList(
+        createCounter(
+            "eclipselink_uow_commits_total",
+            "Total number of unit of work commits",
```

Comment:
Is uow a common term?

Reply:
Afaik, Unit of Work is a common term (see also https://stackoverflow.com/questions/3532258/unit-of-work-pattern-and-history). However, I should probably avoid the abbreviation in the metric name.